File: backend/src/main/java/com/arstem/backend/progress/api/ProgressResponse.java
package com.arstem.backend.progress.api;

import java.util.List;

import com.arstem.backend.progress.domain.Progress;

public record ProgressResponse(String topicCode, int completedSessions, int misconceptionCount, int completionPercent,
        int masteryScore, List<String> weakAreas) {

    public static ProgressResponse from(Progress progress) {
        return new ProgressResponse(progress.getTopicCode(), progress.getCompletedSessions(), progress.getMisconceptionCount(),
                progress.getCompletionPercent(), progress.getMasteryScore(), progress.getWeakAreas());
    }
}

File: frontend/lib/features/progress/models/progress_response.dart
/// Response model for progress endpoints
class ProgressResponse {
  final String topicCode;
  final int completedSessions;
  final int misconceptionCount;
  final int completionPercent;
  final int masteryScore;
  final List<String> weakAreas;

  ProgressResponse({
    required this.topicCode,
    required this.completedSessions,
    required this.misconceptionCount,
    required this.completionPercent,
    required this.masteryScore,
    required this.weakAreas,
  });

  factory ProgressResponse.fromJson(Map<String, dynamic> json) {
    final weakAreas = json['weakAreas'];

    return ProgressResponse(
      topicCode: json['topicCode']?.toString() ?? '',
      completedSessions:
          int.tryParse(json['completedSessions']?.toString() ?? '') ?? 0,
      misconceptionCount:
          int.tryParse(json['misconceptionCount']?.toString() ?? '') ?? 0,
      completionPercent:
          int.tryParse(json['completionPercent']?.toString() ?? '') ?? 0,
      masteryScore: int.tryParse(json['masteryScore']?.toString() ?? '') ?? 0,
      weakAreas: weakAreas is List
          ? weakAreas.map((item) => item.toString()).toList()
          : const [],
    );
  }
}
